File: c++/memory.cpp
#include <string>
#include <iostream>

class Memory : public Hardware{

    //atribut
    private:
        string frequency;
        string memorySize;
        string supportsCuda;

    public:
        //constructor
        Memory(){
            this->frequency = "";
            this ->memorySize = "";
            this ->supportsCuda = "";

        }

        //setter dan getter
        void setFrequency(string frequency){
            this->frequency = frequency;
        }

        string getFrequency(){
            return this->frequency;
        }

        void setMemorySize(string memorySize){
            this->memorySize = memorySize;
        }

        string getMemorySize(){
            return this->memorySize;
        }

        void setSupportsCuda(string supportsCuda){
            this->supportsCuda = supportsCuda;
        }

        string getSupportsCuda(){
            return this->supportsCuda;
        }

        //destructor
        ~Memory(){
        }
};

File: java/Memory.java
class Memory extends Hardware{

    //atribut
    private String frequency;
    private String memorySize;
    private String supportsCuda;

    Memory(){
    }

    //setter dan getter
    public void setFrequency(String frequency){
        this.frequency = frequency;
    }

    public String getFrequency(){
        return this.frequency;
    }

    public void setMemorySize(String memorySize){
        this.memorySize = memorySize;
    }

    public String getMemorySize(){
        return this.memorySize;
    }
    public void setSupportsCuda(String supportsCuda){
        this.supportsCuda = supportsCuda;
    }

    public String getSupportsCuda(){
        return this.supportsCuda;
    }
}